core/src/main/kotlin/com/google/container/tools/core/properties/PropertiesFileFlagReader.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.core.properties
+
+import java.util.Properties
+
+/**
+ * Reads property values from the specified properties file.
+ *
+ * @param propertyFilePath the path to the properties file; defaults to 'config.properties' located
+ * in the root of the module's resources
+ */
+class PropertiesFileFlagReader(
+    propertyFilePath: String = DEFAULT_PROPERTIES_FILE_NAME
+) {
+
+    private val properties: Properties = Properties()
+
+    companion object {
+        private const val DEFAULT_PROPERTIES_FILE_NAME = "/config.properties"
+    }
+
+    init {
+        this.javaClass.getResourceAsStream(propertyFilePath)?.use {
+            properties.load(it)
+        } ?: throw IllegalArgumentException(
```

Comment:
Could we use `log.error` or just throwing an exception will show the error level message anyway? Not sure which one looks like a better fit in here.

Reply:
Yeah. we could presumably do both too (I'll think about it).

But this reminds me of something else. Currently, the way the skaffold process execution works, no exceptions are every bubbled up to the IDE. Instead, when you try to run the process, it shows this little red notification thing (and in this case where I am throwing an exception, the notification will show the exception message). We may want to think about possibly letting certain exceptions bubble through so that we get notified, in the case of truly unanticipated behavior (like failure to load our property file).

Reply:
Agree, good idea. For now as we run system default Skaffold, it's hard to justify IDE level error for most of the run errors - but we definitely should do that once we have more functionality and managed tools and can know more about unexpected errors. We should also think about error notifications too along with console red error messages.

Reply:
Right. In this case however, I think we'd still want to know since its not related directly to the skaffold run (its a side effect of running an analytics ping with an exception loading the properties file). I'll open an issue to track.